From 4b91b4730958cdd454c22223adbea04a50642cd0 Mon Sep 17 00:00:00 2001
From: Abhilash Tuse <Abhilash.Tuse@imgtec.com>
Date: Fri, 25 Nov 2016 15:03:59 +0530
Subject: [PATCH] add SYSLOG_IF_EVERY_N macro support

Signed-off-by: Abhimanyu Vishwakarma <Abhimanyu.Vishwakarma@imgtec.com>
Signed-off-by: Abhilash Tuse <Abhilash.Tuse@imgtec.com>
---
 src/glog/logging.h.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/glog/logging.h.in b/src/glog/logging.h.in
index b1de2c9..5fb14d6 100644
--- a/src/glog/logging.h.in
+++ b/src/glog/logging.h.in
@@ -949,6 +949,9 @@ bool IsFailureSignalHandlerInstalled();
 #define LOG_IF_EVERY_N(severity, condition, n) \
   SOME_KIND_OF_LOG_IF_EVERY_N(severity, (condition), (n), @ac_google_namespace@::LogMessage::SendToLog)
 
+#define SYSLOG_IF_EVERY_N(severity, condition, n) \
+  SOME_KIND_OF_LOG_IF_EVERY_N(severity, (condition), (n), @ac_google_namespace@::LogMessage::SendToSyslogAndLog)
+
 // We want the special COUNTER value available for LOG_EVERY_X()'ed messages
 enum PRIVATE_Counter {COUNTER};
 
-- 
2.6.2

