diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 753d8bd..9721802 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -5,8 +5,8 @@ INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/inc ${CMAKE_SOURCE_DIR}/src)
 INCLUDE(CheckLibraryExists)
 INCLUDE(FindPkgConfig)
 
-PKG_CHECK_MODULES(GLOG ${STRICT_CHECK} libglog)
-
+#PKG_CHECK_MODULES(GLOG ${STRICT_CHECK} libglog)
+SET(GLOG_INCLUDE_DIRS ${STAGING_DIR}/usr/include)
 INCLUDE_DIRECTORIES(${GLOG_INCLUDE_DIRS})
 
 SET(SOURCES main.cc proddata.cc device_data.cc flash_access.cc mtd_access.cc userotp_access.cc vector_operations.cc)
@@ -27,4 +27,4 @@ INSTALL(PROGRAMS wifi_cal.sh DESTINATION bin)
 ######################
 FILE(GLOB CPPLINT_CHECK *.cc *.h)
 LIST(REMOVE_ITEM CPPLINT_CHECK ${CMAKE_CURRENT_SOURCE_DIR}/lib_crc.h)
-CPPLINT_ADD(src ${CPPLINT_CHECK})
+#CPPLINT_ADD(src ${CPPLINT_CHECK})
